### PR TITLE
- removes unecessary publication definitions

### PR DESCRIPTION
**Pull request by baywet**
related #139, CC @AndreaTP

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 env:
   PREVIEW_TASK: publishSnapshotPublicationToSonatypeSnapshotRepository
```
Comment (baywet) [inline .github/workflows/release.yml:10]:
```suggestion
  PREVIEW_TASK: publishMavenPublicationToSonatypeSnapshotRepository
```

**Comment by andreaTP** [comment]
This sounds like a great improvement!
It's not clear to me how those configurations are used ...

**Comment by baywet** [comment]
> It's not clear to me how those configurations are used
if the comment was about where they might be used, here is a bit more context:
- Snapshot is a remnant from msgraph-java-sdk-core, it's effectively only here to add the snapshot suffix, which can be passed by the CLI.
- Maven (the one adding a few properties) is a remnant from when we used to publish to a GitHub packages repository.

neither are needed anymore AFAIK, hence their removal.

**Comment by baywet** [comment]
> use the final version number only for Maven Central releases

> Since you are at it I try, it would be great if the version is computed as follows:
> 
> * always use the suffix `-SNAPSHOT` for development/local releases
> * use the final version number only for Maven Central releases
> 
> This is going to play much better with the rest of the tools/build systems than the current setup.

@andreaTP done, thanks for the suggestion. Let me know if you think we should change anything else

**Review by andreaTP — APPROVED** [review]
LGTM, great cleanup, thanks!

**Pull request merged**